silkworm/common/log.hpp

```diff
@@ -27,31 +28,40 @@ enum LogLevels {
     LogTrace, LogDebug, LogInfo, LogWarn, LogError, LogCrit, LogNone
 };
 
-// Log to two output streams - typically the console and a log file.
+// Log to one or two output streams - typically the console and optional log file.
 class Logger {
 public:
-    Logger(std::ostream& o1, std::ostream& o2, LogLevels level=LogNone)
+    Logger(LogLevels level=LogInfo, std::ostream& o1=std::cerr, std::ostream& o2=Logger::null_stream())
     : stream(o1, o2), verbosity(level) {}
 
     int level(LogLevels level) { return verbosity = level; }
     int level() { return verbosity; }
 
-    std::ostream& log(LogLevels level, const char* file, int line){
-    std::time_t time = std::time(nullptr);
-    return stream << LogTags[level] \
-                  << std::put_time(std::gmtime(&time), " %Y-%m-%d %H:%M:%S %Z ") \
-                  << file << ":" << line << "| ";
+    std::ostream& log(LogLevels level, const char* file, int line) {
+        std::time_t time = std::time(nullptr);
+        return stream << LogTags[level]
+                      << std::put_time(std::gmtime(&time), " %Y-%m-%d %H:%M:%S %Z ")
+                      << file << ":" << line << "| ";
     }
+
+    static Logger& default_logger();
```

Comment:
Where is default_logger() defined? You apparently forgot to add log.cpp into the review.